src/main/java/org/example/TestDataFactory.java

```diff
@@ -1,39 +1,43 @@
 package org.example;
 
 import com.fasterxml.jackson.databind.ObjectMapper;
-import org.apache.commons.lang3.RandomStringUtils;
-import org.apache.hc.client5.http.impl.classic.CloseableHttpResponse;
-import org.apache.hc.core5.http.ClassicHttpResponse;
-import org.apache.hc.core5.http.ParseException;
-import org.apache.hc.core5.http.io.entity.EntityUtils;
+
 import org.example.models.User;
 import org.testng.collections.Lists;
 
 import java.io.IOException;
+import java.net.URISyntaxException;
+import java.net.http.HttpResponse;
 import java.util.List;
 import java.util.Random;
 
 public class TestDataFactory {
 
-    List<String> statusList = Lists.newArrayList("active", "inactive");
+    private String apiURL;
+
+    public TestDataFactory(String apiURL) {
+        this.apiURL = apiURL;
+    }
+
+    private static final List<String> STATUS_LIST = Lists.newArrayList("active", "inactive");
 
-    List<String> genderList = Lists.newArrayList("male", "female");
+    private static final List<String> GENDER_LIST = Lists.newArrayList("male", "female");
 
-    public int getId() throws IOException, ParseException {
+    public int getId() throws IOException, URISyntaxException, InterruptedException {
         ObjectMapper mapper = new ObjectMapper();
-        HttpService httpService = new HttpService();
-        ClassicHttpResponse response = httpService.sendGetRequest(0);
-        User[] users = mapper.readValue(EntityUtils.toString(response.getEntity()), User[].class);
+        HttpService httpService = new HttpService(apiURL);
+        HttpResponse<String> response = httpService.sendGetRequest(0);
+        User[] users = mapper.readValue(response.body(), User[].class);
         return users[0].getId();
     }
 
-    public User generateTestUser() throws IOException, ParseException {
+    public User generateTestUser() throws IOException, URISyntaxException, InterruptedException {
         Random random = new Random();
-        int randomStatusListIndex = random.nextInt(statusList.size());
-        int randomGenderListIndex = random.nextInt(genderList.size());
-        String email = String.format("%s@gmail.com", RandomStringUtils.random(10, true, true));
-        String name = RandomStringUtils.random(7, true, false);
-        User user = new User(getId(), name, email, genderList.get(randomGenderListIndex), statusList.get(randomStatusListIndex));
+        int randomStatusListIndex = random.nextInt(STATUS_LIST.size());
+        int randomGenderListIndex = random.nextInt(GENDER_LIST.size());
+        String email = "TestUser" + System.currentTimeMillis() + "@gmail.com";
+        String name = "TestUser" + System.currentTimeMillis();
+        User user = new User(getId(), name, email, GENDER_LIST.get(randomGenderListIndex), STATUS_LIST.get(randomStatusListIndex));
```

Comment:
If id is a field returned from the API (and not required in the POST request), then create a user without an id.